packages/react-ui-components/src/components/CodeEditor/types.ts

```diff
@@ -0,0 +1,27 @@
+import { editor } from 'monaco-editor';
+import { ReactNode } from 'react';
+
+export type OnEditorChangeCallback = (value: string | undefined) => void;
+export type CustomThemeData = editor.IStandaloneThemeData;
+
+export enum ECodeEditorMode {
```

Comment:
There are no types / enums we can extend from, having checked the [docs](https://microsoft.github.io/monaco-editor/api/modules/monaco.languages.html) and the [repo](https://github.com/microsoft/monaco-editor/tree/main/src/basic-languages) I didn't find any, I'll leave this open for discussion @djcouto.

Reply:
I don't think "monaco editor" has an enum or type with all its supported languages. IMO and based on the fact that they accept any string as language, we should forward the component language prop to the editor component as a string without validations. If the editor doesn't support the language passed as a string, it will simply not be highlighted and won't raise an error.